liquidity_pool/src/utils.rs

```diff
@@ -95,16 +95,17 @@ pub trait UtilsModule:
     fn get_capital_utilisation(&self) -> BigUint {
         let reserve_amount = self.reserves().get();
         let borrowed_amount = self.borrowed_amount().get();
+        let rewards_reserves_paid = self.rewards_reserves_paid().get();
+        let total_amount = &reserve_amount + &borrowed_amount - rewards_reserves_paid;
```

Comment:
I was trying to review the design, but I keep getting stuck when trying to figure out `self.reserves()` and `self.rewards_reserves_paid()`, and this is a good place to ask about them.

So, in principle, `self.reserves()` seems to be equal to the deposited amount, plus the interest for that deposited amount (which is equal to the total interest for the borrowed amount since the contract was created, that wasn't yet distributed to lenders through the `withdraw` endpoint), minus the borrowed amount. Is that correct?

On the other hand, `self.rewards_reserves_paid()` seems to be the total amount of interest ever paid by the borrowers, including the amounts that were already distributed to lenders (e.g. the `withdraw` endpoint does not seem to change it). Is that correct?

If so, then why does it make sense to subtract `rewards_reserves_paid` from `reserve_amount + borrowed_amount`?

One option would be that the intent was to make `self.rewards_reserves_paid()` equal to the total interest for the borrowed amount that wasn't yet distributed to lenders through the `withdraw` endpoint. Is that the case?

Assuming that, then why is `rewards_reserves_paid` subtracted here? When borrowing, users can borrow up to the entire `self.reserves()`, which means that capital utilization can be larger than 100% (i.e. it's possible that `self.reserves() == 0`, which means that `utilization = borrowed_amount / (borrowed_amount - rewards_reserves_paid)`). Is that intentional?

Reply:
Yep, you are correct. `self.reserves()` is equal to deposited amount, plus the interest for that deposited amount minus the borrowed amount.

Yes, `self.rewards_reserves_paid()` is the total amount of interest ever paid. It is decoupled of lenders. It is a variable only modified by the borrowers.

Well, the [reserves are increased](https://github.com/ElrondNetwork/elrond-lend-rs/blob/f2fcc4eab730cb9d6d90fb486823f9d9dfb3139b/liquidity_pool/src/liquidity.rs#L294) at repayment. We must substract the `rewards_reserves_paid` to get only the original deposited capital. We don't want to have compounded interest.

Indeed, `self.rewards_reserves_paid()` is a variable that equals to the total interest for the borrowed amount that wasn't yet distributed to lenders. I must adjust it at `withdraw` scenario and make a unit test for it.

Reply:
Asking to make sure: Is it fine to have a utilization that is larger than 100%?

Reply:
Nope, that's an issue. I will create an unit test and try to catch the scenario you described.

Reply:
@virgil-serbanuta ,

Yes, 

> `self.rewards_reserves_paid()` equals to the total interest for the borrowed amount that wasn't yet distributed to lenders through the withdraw endpoint.

I fixed this by substracting the interest in the `withdraw` endpoint. Thanks!

Regarding the issue:

> Assuming that, then why is rewards_reserves_paid subtracted here? When borrowing, users can borrow up to the entire self.reserves(), which means that capital utilization can be larger than 100% (i.e. it's possible that self.reserves() == 0, which means that utilization = borrowed_amount / (borrowed_amount - rewards_reserves_paid)). Is that intentional?

This will never be the case. `self.reserves()` will be 0 only if `self.rewards_reserves_paid()` will be 0. Because, at any time, `self.reserves()` contains the `self.rewards_reserves_paid()`, rewards that were not distributed to suppliers.

I am substracting because, in the total value available, I am not considering the interest that was accumulated in the contract and it is owed to the suppliers as availalbe.

Therefore, I:
* renamed the variable:  `rewards_reserves_paid` -> `rewards_reserves_accumulated_not_distributed`
* adjusted the value in `withdraw` endpoint: substracted the interest.